test/blackbox-tests/test-cases/pkg/pin-depends.t

```diff
@@ -175,10 +175,18 @@ Pin to an invalid opam file
   unsupported or missing file format version; should be 2.0 or older
   [1]
 
-Pin to an HTTP archive doesn't work
+Pin to an HTTP archive work
 
-  $ runtest "http://0.0.0.0/tarball.tgz"
-  File "foo.opam", line 1, characters 0-0:
-  Error: Could not determine location of repository http://0.0.0.0/tarball.tgz
-  Hint: Specify either a file path or git repo via SSH/HTTPS
-  [1]
+  $ mkdir _source/
+  $ cat > _source/bar.opam << EOF
+  > opam-version: "2.0"
+  > EOF
+  $ tar -czf tarball.tgz -C _source bar.opam
+  $ echo tarball.tgz > fake-curls
+  $ PORT=1
+  $ runtest "http://0.0.0.0:$PORT/tarball.tgz"
+  Solution for dune.lock:
+  - bar.1.0.0
+  (version 1.0.0)
+  (dev)
+  (source (fetch (url http://0.0.0.0:1/tarball.tgz) (checksum md5=ecfb12cd56789e48aaa4eef289c85c40))) (dev) 
```

Comment:
I think it would be nice to determine the MD5 of the tarball that was created and make sure that it matches the one in the lock file. Otherwise it's not reproducible.

Reply:
I have do it using grep as the tar headers can make the hash different.